cmsl1t/collections/base.py

```diff
@@ -40,20 +40,20 @@ def create_n_dim_dict(dimensions, initiaValue=0):
 
 def len_n_dim_dict(dictionary, dimensions):
     if dimensions <= 1:
-        return len(dictionary)
+        return len(dictionary.keys())
```

Comment:
Do you need the `keys()` bit?  Doesn't seem "pythonic" to me

Reply:
Yeah, that is a workaround due to having `BaseHistCollection.__len__` which can be the `dictionary` in the above function. Without `.keys()` you get infinite recursion